api/stripe.go

```diff
@@ -194,34 +208,35 @@ func (api *API) processStripeSub(ctx context.Context, s *stripe.Subscription) er
 		Customer: s.Customer.ID,
 		Status:   string(stripe.SubscriptionStatusActive),
 	})
+	// TODO Allow multiple stripe ids per user?
 	u, err := api.staticDB.UserByStripeID(ctx, s.Customer.ID)
 	if err != nil {
 		return errors.AddContext(err, "failed to fetch user from DB based on subscription info")
 	}
-	// Set the default values.
-	u.Tier = database.TierFree
-	u.SubscribedUntil = time.Time{}
-	u.SubscriptionStatus = ""
-	u.SubscriptionCancelAt = time.Time{}
-	u.SubscriptionCancelAtPeriodEnd = false
-	var latestCreated time.Time
-	var latestSub string
 	// Pick the latest active plan and set the user's tier based on that.
 	subs := it.SubscriptionList().Data
+	var mostRecentSub *stripe.Subscription
 	for _, subsc := range subs {
-		created := time.Unix(subsc.Created, 0).UTC()
-		if created.After(latestCreated) {
-			latestCreated = created
-			latestSub = subsc.ID
-			// It seems weird that the Plan.ID is actually a price id but this
-			// is what we get from Stripe.
-			u.Tier = stripePrices()[subsc.Plan.ID]
-			u.SubscribedUntil = time.Unix(subsc.CurrentPeriodEnd, 0).UTC()
-			u.SubscriptionStatus = string(subsc.Status)
-			u.SubscriptionCancelAt = time.Unix(subsc.CancelAt, 0)
-			u.SubscriptionCancelAtPeriodEnd = subsc.CancelAtPeriodEnd
+		if mostRecentSub == nil || mostRecentSub.Created < subsc.Created {
```

Comment:
(nit) `if mostRecentSub == nil || subsc.Created > mostRecentSub.Created {` is more logical

Reply:
Sure, makes sense.